src/Microsoft.OData.Core/MediaTypeUtils.cs

```diff
@@ -217,9 +217,12 @@ internal static Encoding MissingEncoding
 
             ODataFormat format = GetFormatFromContentType(contentTypeHeader, supportedPayloadKinds, mediaTypeResolver, out mediaType, out encoding, out selectedPayloadKind);
 
-            // for batch payloads, read the batch boundary from the content type header; this is the only
-            // content type parameter we support (and that is required for batch payloads)
-            if (selectedPayloadKind == ODataPayloadKind.Batch)
+            batchBoundary = null;
+
+            // For batch payload of multipart/mixed type, read the required batch boundary parameter from the content type header.
+            if (selectedPayloadKind == ODataPayloadKind.Batch &&
+                !(HttpUtils.CompareMediaTypeNames(MimeConstants.MimeApplicationType, mediaType.Type) &&
+                  HttpUtils.CompareMediaTypeNames(MimeConstants.MimeJsonSubType, mediaType.SubType)))

```

Comment:
Rather than checking if it's batch && !(application/json), we should check if it's batch && multipart/mixed.  The boundary assumes multipart-mixed, and if we added another media type in the future we would have to explicitly exclude it here as well.